core-java-sample/build.gradle

```diff
@@ -9,9 +9,7 @@ dependencies {
     testCompile group: 'junit', name: 'junit', version: '4.12'
     testCompile project(path: ":core-java", configuration: 'testArtifacts')
 
-    compile project(':core-java')
-    compile project(':java-gae')
-
+    compile project(path: ':core-java');
     compile 'commons-io:commons-io:2.4'
```

Comment:
org.apache.commons.io.FileUtils.deleteDirectory method is used in FileSystemStorageFactory. Unfortunately, there is no such method in java.nio (its delete method can delete only empty dirs).